docs/proposals/caching.adoc

```diff
@@ -0,0 +1,77 @@
+Also in a future version the `pipeline-cache-dir` may be on a different PVC as the checkout PVC for example to enable faster cleanup of the pristine checkout dir by simply recreating it or not even having it on a PVC. 
+
+Files in the cache locations would persist between builds unless they are removed by the cleanup mechanism. 
+
+The build scripts are adjusted to take advantage of the cache parameters above. In addition each build task will support a new parameters `global-caching-mb` and `pipeline-caching-mb`. These defaults to 0 to indicate the task is not using caching. Otherwise it specify the minimum required cache space in MB.    
+
+All build scripts are adjusted to log
+
+- the time longer running commands take. 
+
+Furthermore the available disks pace is be logged as well (not sure where this should be done at the moment)
+
+=== ods-start and cleanup
+
+Build tasks can rely on that cache cleanup does not occur while they are running as well as that these would not be partially cleaned up. 
+
+On the other hand build Tasks cannot rely on that a cache remains populated from a previous build.
+
+The cleanup strategy described here is subject to change even in patch versions and specific details must not be relied on.
+
+Pipeline cleanup happens during `ods-start`, so that no other build can run concurrently.
+
+Before cache cleanup `ods-start` cleans up the prior PVC checkout dir, which means everything except `/.cache/<some-dir>` or `/.cache-p/<some-dir>`. In particular to prevent having files at the top levels regular files or links below `/.cache/` or `/.cache-p/` are deleted.  
+
+`ods-start` then cleans up separately the global and pipeline cache to fulfill the declared requirements of all build tasks in the odl.yaml. If disk space remains too low in a particular cache area a warning message is logged and the respective parameter `global-cache-parent` and/or `pipeline-cache-dir` will be empty. 
+
+
+== Pro
+
+* Caching enabled for tools using global cache
+
+* Caching enabled where it is best to build in a cached context per pipeline
+
+* Opt-in to caching can be used to document best practices for PVC sizing.
+
+* By having cleanup done in ods-start no need to have an interleaved cleanup process.
+
+== Con
+
+* Tools such as for typescript https://github.com/pnpm/pnpm[pnpm] and https://github.com/yarnpkg/berry[yarn berry] support deduplication allowing to reference artifacts from a central location without packaging them again. pnpn uses hard links to make this efficient. However hard linking is not supported across PVCs and thus could not be used with the proposed design as is. See also related discussion at https://github.com/opendevstack/ods-pipeline/discussions/411[node typescript security and caching - npm or yarn? #411].
```

Comment:
To enable exploring the newer npm techs using hard linking, I am now in favor of changing the proposal so that `pipeline-cache-dir` and `global-cache-parent` would always be on the same PVC. However the checkout dir might still not be on the same PVC and perhaps not even be a PVC at all.